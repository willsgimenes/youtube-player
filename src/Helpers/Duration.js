export const prettyDuration = time => {
  const string = time.match(/(\d+)(?=[MHS])/ig) || []

  return string.map(function (item) {
    if (item.length < 2) return '0' + item
    return item
  }).join(':')
}

export const converToSeconds = time => {
  const string = time.match(/(\d+)(?=[MHS])/ig) || []

  const parsed = string.map(function (item) {
    if (item.length < 2) return '0' + item
    return item
  }).join(':')

  return parsed.split(':').reduce((acc, time) => (60 * acc) + +time)
}
