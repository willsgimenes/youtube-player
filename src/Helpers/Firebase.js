import Axios from 'axios'

const BASE_URL = 'https://f-player.herokuapp.com/'

export const fetchIndex = async () => {
  return Axios.get(`${BASE_URL}songs/index`)
}

export const fetchSongs = async () => {
  return Axios.get(`${BASE_URL}songs/list`)
}

export const postList = async song => {
  return Axios.post(`${BASE_URL}songs`, song)
}
