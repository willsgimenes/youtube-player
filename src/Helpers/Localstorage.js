/* global localStorage */

export const loadState = (state) => {
  try {
    const serializedState = localStorage.getItem(state)

    if (serializedState === null) {
      return undefined
    }
    return JSON.parse(serializedState)
  } catch (err) {
    return {}
  }
}

export const loadStateNotSerialized = (state) => {
  const serializedState = localStorage.getItem(state)
  if (serializedState === null) {
    return undefined
  }
  return serializedState
}

export const saveState = (state, value) => {
  let serializedState = value

  if (typeof serializedState === 'string') {
    localStorage.setItem(state, serializedState)
  } else {
    serializedState = JSON.stringify(value)
    localStorage.setItem(state, serializedState)
  }
}

export const removeItem = (key) => {
  localStorage.removeItem(key)
}
