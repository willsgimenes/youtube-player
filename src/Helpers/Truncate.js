export default function (string, length) {
  var truncatedString = string

  if (string.length >= length) { truncatedString = string.substring(0, length - 3) + '...' }

  return truncatedString
}
