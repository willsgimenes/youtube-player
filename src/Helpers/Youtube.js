import Axios from 'axios'

const fetchVideo = async term => {
  return Axios.get('https://f-player.herokuapp.com/', {
    params: {
      term: term
    }
  })
}

const getVideoByTitle = (term) => {
  return new Promise((resolve, reject) => {
    fetchVideo(term)
      .then(({ data }) => resolve(data))
      .catch(error => reject(error))
  })
}

export { getVideoByTitle, fetchVideo }
