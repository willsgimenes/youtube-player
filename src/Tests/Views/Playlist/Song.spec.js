/* global describe */
/* global beforeEach */
/* global it */
/* global expect */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import SongComponent from '../../../Views/Playlist/Song'

import { saveState } from '../../../Helpers/Localstorage'

function storageMock () {
  let storage = {}

  return {
    setItem: function (key, value) {
      storage[key] = value || ''
    },
    getItem: function (key) {
      return key in storage ? storage[key] : null
    },
    removeItem: function (key) {
      delete storage[key]
    },
    get length () {
      return Object.keys(storage).length
    },
    key: function (i) {
      let keys = Object.keys(storage)
      return keys[i] || null
    }
  }
}

window.localStorage = storageMock()

saveState(
  'StreamItems-ac0d862c-d702-4b13-7150-5f982a590f98',
  {
    id: 'ac0d862c-d702-4b13-7150-5f982a590f98',
    title: 'foobar',
    song: {
      id: '123',
      type: 'youtube',
      duration: 123,
      prettyDuration: 123,
      author: 'william',
      url: 'youtube',
      cleanTitle: 'test video'
    }
  }
)

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

Enzyme.configure({ adapter: new Adapter() })

describe('Song Component', () => {
  let
    wrapper,
    store,
    initialState

  beforeEach(() => {
    initialState = {
      player: {
        index: 'a3e9bf26-903f-e050-3879-c3f5f8271efd',
        items: [
          'ac0d862c-d702-4b13-7150-5f982a590f98',
          'a3e9bf26-903f-e050-3879-c3f5f8271efd',
          '8e5cde91-5170-f6fd-b210-eeda77e5a00e'
        ],
        prev: 'ac0d862c-d702-4b13-7150-5f982a590f98',
        next: '8e5cde91-5170-f6fd-b210-eeda77e5a00e',
        song: {
          '8e5cde91-5170-f6fd-b210-eeda77e5a00e': {
            'song': {
              'id': 'dr3md9saDwk',
              'type': 'youtube',
              'title': 'Major S4 Ame Nochi Niji Iro with lyrics',
              'duration': 242,
              'author': 'Sun Tatsunari',
              'prettyDuration': '04:02',
              'url': 'https://www.youtube.com/watch?v=dr3md9saDwk',
              'cleanTitle': 'Major S4 Ame Nochi Niji Iro with lyrics'
            },
            'title': 'Major S4 Ame Nochi Niji Iro with lyrics',
            'id': '8e5cde91-5170-f6fd-b210-eeda77e5a00e'
          }
        }
      }
    }

    store = mockStore(initialState)
    wrapper = shallow(<SongComponent
      store={store}
      item='8e5cde91-5170-f6fd-b210-eeda77e5a00e'
      index='8e5cde91-5170-f6fd-b210-eeda77e5a00e'
    />)
  })

  it('should render properly', () => {
    expect(wrapper.dive().find('Song')).toHaveLength(1)
    expect(wrapper).toMatchSnapshot()
  })

  it('should not render if hasn\'t any song', () => {
    const newInitialState = {
      player: {
      }
    }

    store = mockStore(newInitialState)
    wrapper = shallow(<SongComponent
      store={store}
    />)

    expect(wrapper.dive().find('Song')).toHaveLength(0)
    expect(wrapper).toMatchSnapshot()
  })

  it('should test handleClick', () => {
    wrapper.dive().find('Song').props().handleClick({ id: '8e5cde91-5170-f6fd-b210-eeda77e5a00e' })
  })
})
