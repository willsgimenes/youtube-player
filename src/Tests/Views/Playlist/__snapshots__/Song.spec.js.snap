// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song Component should not render if hasn't any song 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(SongComponent)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "changeSong": [Function],
      "song": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "changeSong": [Function],
        "song": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "storeSubscription": undefined,
  },
}
`;

exports[`Song Component should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(SongComponent)
    index="8e5cde91-5170-f6fd-b210-eeda77e5a00e"
    item="8e5cde91-5170-f6fd-b210-eeda77e5a00e"
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "changeSong": [Function],
      "index": "8e5cde91-5170-f6fd-b210-eeda77e5a00e",
      "item": "8e5cde91-5170-f6fd-b210-eeda77e5a00e",
      "song": Object {
        "8e5cde91-5170-f6fd-b210-eeda77e5a00e": Object {
          "id": "8e5cde91-5170-f6fd-b210-eeda77e5a00e",
          "song": Object {
            "author": "Sun Tatsunari",
            "cleanTitle": "Major S4 Ame Nochi Niji Iro with lyrics",
            "duration": 242,
            "id": "dr3md9saDwk",
            "prettyDuration": "04:02",
            "title": "Major S4 Ame Nochi Niji Iro with lyrics",
            "type": "youtube",
            "url": "https://www.youtube.com/watch?v=dr3md9saDwk",
          },
          "title": "Major S4 Ame Nochi Niji Iro with lyrics",
        },
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "changeSong": [Function],
        "index": "8e5cde91-5170-f6fd-b210-eeda77e5a00e",
        "item": "8e5cde91-5170-f6fd-b210-eeda77e5a00e",
        "song": Object {
          "8e5cde91-5170-f6fd-b210-eeda77e5a00e": Object {
            "id": "8e5cde91-5170-f6fd-b210-eeda77e5a00e",
            "song": Object {
              "author": "Sun Tatsunari",
              "cleanTitle": "Major S4 Ame Nochi Niji Iro with lyrics",
              "duration": 242,
              "id": "dr3md9saDwk",
              "prettyDuration": "04:02",
              "title": "Major S4 Ame Nochi Niji Iro with lyrics",
              "type": "youtube",
              "url": "https://www.youtube.com/watch?v=dr3md9saDwk",
            },
            "title": "Major S4 Ame Nochi Niji Iro with lyrics",
          },
        },
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "storeSubscription": undefined,
  },
}
`;
